### DISCUSSION TITLE
Integration between kepler and autoscalers inside kubernetes

### DISCUSSION THREAD

**Discussion by thisiskazem** [sustainable-computing-io/kepler#1380] [General]
Is there any possible solution to integrate kepler with the horizontal and vertical autoscalers inside kubernetes to be able to automate scaling up and down the objects inside the kubernetes according to the kepler metrics?
To scale up  a web application deployment horizontally or vertically while the power consumption metrics increase or decrease as an example.

**Comment by zhifanl** [comment]
^^ I want to do the same thing.

**Comment by zhifanl** [comment]
Can we use the carbon-aware Keda operator to scale it automatically? Link: https://github.com/Azure/carbon-aware-keda-operator?tab=readme-ov-file
and instead of exporting the carbon intensity from watttime API, we export carbon emission data by querying the prom metric and put it in the ConfigMap.
